### PR TITLE
Join a channel only if invited

### PR DESCRIPTION
**Pull request by vfurmane**
Merge after #123

### CODE REVIEW

File: include/IRCErrors.hpp [hunk 1]
```diff
@@ -216,6 +216,20 @@ class ERR_UNKNOWNMODE : public AIRCError
...
+
+	public:
+		ERR_INVITEONLYCHAN(const std::string &nickname, const std::string &channel) : _str("473 " + nickname + " " + channel + " :Cannot join channel (+i)") {}
+		~ERR_INVITEONLYCHAN(void) throw() {}
+		virtual const char* what() const throw()
+		{
+			return this->_str.c_str();
+		}
+};
+
```
Comment (ppiques) [inline include/IRCErrors.hpp:232]: should be a separate PR

**Review by theovgl — APPROVED** [review]
